### ISSUE TITLE
aws_ssm_parameter resource does not have a overwrite option

### ISSUE THREAD

**Issue by FransUrbo** [terraform-providers/terraform-provider-aws#1004]
I'm trying to update/replace a *Parameter Store* value, but TF gives `ParameterAlreadyExists`.

### Terraform Version
0.9.8

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_ssm_parameter

**Comment by FransUrbo** [comment]
I notice after I've deleted the param, run `terraform apply` and then ran `terraform apply` again some time later, that it can't detect that the parameter already exists!

So a better option (because I'm encrypting the parameter with a KMS key and I'm not sure TF can get the param, decrypt it with the key - how would it know WHICH key?? - and then check if it differs from what is in the state or the `value` option) might be to simply try see if the param already exists, and then ignore it (as in, not to create one).

**Comment by FransUrbo** [comment]
Another issue:

If the `name` value is changed (not sure if it's related to the param also been deleted), I get:
```
[ERROR] SSM Parameter XXX is invalid
```

**Comment by stack72** [comment]
@FransUrbo if possible, can you try and recreate that issue and then open an alternative about that?

**Comment by FransUrbo** [comment]
Ok, no worries!

Regarding the "can't detect already exists" above. This MIGHT be because I had the same resource name (but different `name` value) in several modules.

As in:
```
resource "aws_ssm_parameter" "hiera-private" {
    [...]
}
```
in all my modules. Prefixing the resource name, seems to have solved the "recreation issue"...

**Comment by FransUrbo** [comment]
PS. Please back port to 0.9 if possible!

**Comment by stack72** [comment]
Hi @FransUrbo 

there will be no further releases of the 0.9 services I'm afraid

Paul

**Comment by FransUrbo** [comment]
Ok, bummer :(. So 0.10 is starting to be ok? Last I saw it was just a beta release... ?

**Comment by stack72** [comment]
@FransUrbo it was the first release in the major version series - it will take a few betas before we GA it

**Comment by FransUrbo** [comment]
If that's so, I'd very much like you to continue supporting 0.9. I'm not ready (i.e., brave/stupid enough), to hand my whole production environment to beta code!

**Comment by stack72** [comment]
Sorry, we are only making critical big fixes to 0.9.x series. Ann
development effort is focused on 0.10 series now

It will be ready for GA in a few weeks

Paul

**Comment by FransUrbo** [comment]
That's kind'a what I meant by "support 0.9". I did NOT mean to add features etc into it so we're in full agreement there. However, we might have a slight disagreement on the word *critical* :).

This one does not comply to that IMO, just to make things clear :D.
